ui/src/main/webapp/admin-tools/wcpm/policy.js

```diff
@@ -0,0 +1,152 @@
+import AuthTypes, { AuthTypesView } from './auth-types'
+import Realms, { RealmsView } from './realms'
+import RequiredClaims, { RequiredClaimsView } from './required-claims'
+
+const Layout = ({ children, number }) => (
+  <div>
+    <div style={{ padding: 8 }}>
+      <Header>Policy #{number}</Header>
+    </div>
+    {children}
+  </div>
+)
+
+const PolicyEdit = (props) => {
+  const {
+    data: {
+      loading,
+      sts = { claims: [] },
+      wcpm = { realms: [], authTypes: [] }
+    },
+
+    policy,
+    errors = {},
+    serverErrors = {},
+    number,
+
+    onUpdate,
+    onCancel,
+    onSave,
+    onDelete
+  } = props
+
+  return (
+    <Layout number={number}>
+      <Spinner submitting={loading}>
+        <ContextPaths
+          paths={policy.paths}
+          errors={errors.paths}
+          onUpdate={(value, path = []) => onUpdate(['paths'].concat(path), value)} />
+        <ServerErrors errors={serverErrors.paths} />
+        <Divider />
+        <AuthTypes
+          wcpm={wcpm}
+          authTypes={policy.authTypes}
+          errors={errors.authTypes} onUpdate={(value, path = []) => onUpdate(['authTypes'].concat(path), value)} />
+        <ServerErrors errors={serverErrors.authTypes} />
+        <Divider />
+        <Realms
+          wcpm={wcpm}
+          realm={policy.realm}
+          onUpdate={(value, path = []) => onUpdate(['realm'].concat(path), value)} />
+        <ServerErrors errors={serverErrors.realm} />
+        <Divider />
+        <RequiredClaims
+          sts={sts}
+          claimsMapping={policy.claimsMapping}
+          onUpdate={(value, path = []) => onUpdate(['claimsMapping'].concat(path), value)} />
+        <ServerErrors errors={serverErrors.claimsMapping} />
+        <ConfirmationPanel
+          onCancel={onCancel}
+          onSave={() => { onSave(policy) }}
+          allowDelete
+          onDelete={onDelete} />
+        <ServerErrors errors={serverErrors.all} />
+      </Spinner>
+    </Layout>
+  )
+}
+
+PolicyEdit.fragments = {
+  policyEdit: gql`
+    fragment policyEdit on Query {
+      sts { claims }
+      wcpm {
+        realms
+        authTypes
+      }
+    }
+  `
+}
+
+const PolicyView = ({ onEdit, policy, number }) => (
+  <EditRegion onEdit={() => onEdit(createPolicy(policy))} >
+    <Layout number={number}>
+      <ContentPathsView paths={policy.paths} />
+      <Divider />
+      <AuthTypesView authTypes={policy.authTypes} />
+      <Divider />
+      <RealmsView realm={policy.realm} />
+      <Divider />
+      <RequiredClaimsView claimsMapping={policy.claimsMapping} />
+    </Layout>
+  </EditRegion>
+)
+
+PolicyView.fragments = {
+  policy: gql`
+    fragment policy on ContextPolicyBinPayload {
+      paths
+      authTypes
+      realm
+      claimsMapping {
+        key
+        value
+      }
+    }
+  `
+}
+
+export { PolicyEdit }
+export default PolicyView
+
+const Policy = Record({
+  realm: 'karaf',
```

Comment:
It's entirely possible that the backend is not in a state that could support the karaf realm. For example, by deleting `user.attributes` or `user.properties` file in ddf home, no one would have access to the system. This scenario is entirely possible since it's a step in the hardening guide, so we should not display this option to the user if the backend cannot support it.